atoma-confidential/src/sev_snp.rs

```diff
@@ -0,0 +1,138 @@
+use crate::ToBytes;
+
+use sev::{
+    attestation::{AttestationReport, AttestationReportSignature, Chain},
+    error::SevError,
+    firmware::Firmware,
+};
+
+use thiserror::Error;
+
+pub const SEV_SNP_REPORT_DATA_SIZE: usize = 64;
+
+type Result<T> = std::result::Result<T, SevError>;
+
+/// Generates an SEV-SNP attestation report for the given compute data.
+///
+/// This function takes a slice of bytes representing the attested data and returns
+/// an AttestationReport. The attested data must be exactly 64 bytes long,
+/// matching the SEV_SNP_REPORT_DATA_SIZE constant.
+///
+/// # Arguments
+///
+/// * `attested_data` - A slice of bytes containing the data to be attested (public key)
+///
+/// # Returns
+///
+/// * `Result<AttestationReport>` - On success, returns an AttestationReport.
+///                                 On failure, returns a SevError.
+///
+/// # Errors
+///
+/// Returns `SevError::InvalidAttestedDataSize` if the input data size is not 64 bytes.
+/// Returns `SevError::FailedToOpenFirmware` if the firmware interface cannot be opened.
+/// Returns `SevError::FailedToGetAttestationReport` if the attestation report cannot be generated.
+///
+/// # Example
+///
+/// ```
+/// let data_to_attest = [1u8; 64];
+/// match get_compute_data_attestation(&data_to_attest) {
+///     Ok(report) => println!("Attestation successful"),
+///     Err(e) => eprintln!("Attestation failed: {}", e),
+/// }
+/// ```
+pub fn get_compute_data_attestation(attested_data: &[u8]) -> Result<SNPAttestationReport> {
```

Comment:
This looks great to me, we do need to look further how to incorporate NVIDIA cc into these remote attestation generation.

Reply:
Yes, nvtrust port in progress.